client/post-editor/editor-featured-image/index.jsx

```diff
@@ -120,7 +121,11 @@ class EditorFeaturedImage extends Component {
 
 		return (
 			<div className={ classes }>
-				{ site && featuredImageId && <QueryMedia siteId={ site.ID } mediaId={ featuredImageId } /> }
+				{
+					site && featuredImageId && isNumber( featuredImageId )
```

Comment:
The `QueryMedia` component expects a numeric image ID when querying for media. While the featured image is uploaded, a temporary transient image is used and its ID is in the form `featured-image<n>`, which is not numeric. 

Full error: 
```
Warning: Failed prop type: Invalid prop `mediaId` of type `string` supplied to `QueryMedia`, expected `number`.
```

Also while the transient is used, the image doesn't actually exist yet on the server side, which means an extra failed request is done for it.

The `isNumber` check makes sure the image is created on the backend, before it's queried by `QueryMedia`